interoperator/controllers/schedulers/updateinstancecount.go

```diff
@@ -0,0 +1,110 @@
+//go:build schedulers
+// +build schedulers
+
+/*
+Copyright 2018 The Service Fabrik Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schedulers
+
+import (
+	"context"
+
+	osbv1alpha1 "github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/api/osb/v1alpha1"
+	resourcev1alpha1 "github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/api/resource/v1alpha1"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/constants"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	kubernetes "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var log = ctrl.Log.WithName("updateInstanceCount")
+
+func updateInstanceCount(kubeConfig *rest.Config, scheme *runtime.Scheme, mapper meta.RESTMapper) {
+
+	if kubeConfig == nil {
+		err := errors.NewInputError("updateInstanceCount", "kubeConfig", nil)
+		log.Error(err, "invalid input")
+	}
+
+	if scheme == nil {
+		err := errors.NewInputError("updateInstanceCount", "scheme", nil)
+		log.Error(err, "invalid input")
+	}
+
+	err := resourcev1alpha1.AddToScheme(scheme)
+	if err != nil {
+		log.Error(err, "failed to create k8s client")
+	}
+
+	client, err := kubernetes.New(kubeConfig, kubernetes.Options{
+		Scheme: scheme,
+		Mapper: mapper,
+	})
+	if err != nil {
+		log.Error(err, "failed to create k8s client")
+	}
+
+	ctx := context.Background()
+	sfserviceinstances := &osbv1alpha1.SFServiceInstanceList{}
+	instanceOptions := &kubernetes.ListOptions{}
+
+	// Calculate the expected service instance count for each cluster
+	instanceCount := make(map[string]int)
+	for more := true; more; more = (sfserviceinstances.Continue != "") {
+		err := client.List(ctx, sfserviceinstances, instanceOptions, kubernetes.Limit(constants.ListPaginationLimit),
```

Comment:
It is nice that we are only reading 100 (constants.ListPaginationLimit) sfserviceinstances at a time 👍 . This should prevent the possibility of an out of memory error in heavy landscapes. I think it will be a good idea to monitor memory usage after releasing this feature.
Here is a guide from Golang garbage collector, https://tip.golang.org/doc/gc-guide#Optimization_guide